### PR TITLE
plan: eliminate tablescan's cost more precisely.

### PR DESCRIPTION
**Pull request by hanfei1991**
We forget to consider cpu cost when the table scan has a condition.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: plan/physical_plans.go [hunk 1]
```diff
@@ -65,26 +65,30 @@ type physicalDistSQLPlan interface {
 	addAggregation(ctx context.Context, agg *PhysicalAggregation) expression.Schema
 	addTopN(ctx context.Context, prop *requiredProperty) bool
 	addLimit(limit *Limit)
-	calculateCost(count uint64) float64
+	calculateCost(count uint64, scanCount uint64) float64
```
Comment (shenli) [inline plan/physical_plans.go]: Add comment for the two parameters.

Comment (shenli) [inline plan/physical_plans.go]: Should we rename count to resultCount?

File: plan/physical_plans.go [hunk 2]
```diff
@@ -65,26 +65,30 @@ type physicalDistSQLPlan interface {
...
 }
 
-func (p *PhysicalIndexScan) calculateCost(count uint64) float64 {
-	cnt := float64(count)
+func (p *PhysicalIndexScan) calculateCost(resultCount uint64, scanCount uint64) float64 {
```
Comment (coocood) [inline plan/physical_plans.go]: Why scanCount is not used here?

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**